memorystore/redis/cloud_run_deployment/e2e_test.py

```diff
@@ -116,46 +74,85 @@ def services():
             "--tag",
             image_name,
             "--project",
-            project,
+            PROJECT,
         ], check=True
     )
+    yield image_name
+
     subprocess.run(["rm", "Dockerfile"], check=True)
 
+    # Delete container image
+    subprocess.run(
+        [
+            "gcloud",
+            "container",
+            "images",
+            "delete",
+            image_name,
+            "--quiet",
+            "--project",
+            PROJECT,
+        ], check=True
+    )
+
+
+@pytest.fixture
+def deployed_service(container_image, redis_host):
     # Deploy image to Cloud Run
-    service_name = f"test-visit-count-{suffix}"
+    service_name = f"test-visit-count-{SUFFIX}"
     subprocess.run(
         [
             "gcloud",
             "run",
             "deploy",
             service_name,
             "--image",
-            image_name,
+            container_image,
             "--platform=managed",
             "--no-allow-unauthenticated",
             "--region=us-central1",
             "--vpc-connector",
-            connector_name,
+            VPC_CONNECTOR_NAME,
             "--set-env-vars",
             f"REDISHOST={redis_host},REDISPORT=6379",
             "--project",
-            project,
+            PROJECT,
         ], check=True
     )
+    yield service_name
 
+    # Delete Cloud Run service
+    subprocess.run(
+        [
+            "gcloud",
+            "run",
+            "services",
+            "delete",
+            service_name,
+            "--platform=managed",
+            "--region=us-central1",
+            "--quiet",
+            "--project",
+            PROJECT,
+        ], check=True
+    )
+
+
+@pytest.fixture
+def service_url_auth_token(deployed_service):
```

Comment:
I like the way you separated out each part of the "services" into their own fixtures!